pre-sync/oci-image-verification/README.md

```diff
@@ -0,0 +1,139 @@
+-keyout tls.key \
+-out tls.crt \
+-days 356 \
+-subj "/CN=webhook-service.oci-webhook.svc"  \
+-addext "subjectAltName = DNS:webhook-service,DNS:webhook-service.oci-webhook.svc,DNS:webhook-service.oci-webhook"
+```
+
+Create on cluster secret:
+
+```angular2html
+kubectl create secret tls webhook-tls --cert=tls.crt --key=tls.key -n oci-webhook
+```
+
+### IAM setup for the Webhook Server
+
+- Give the Google service account associated with the webhook server permission to read images from Artifact Registry
+```angular2html
+gcloud artifacts repositories add-iam-policy-binding <AR_REPO> \
+   --location=<LOCATION> \
+   --member=serviceAccount:<GSA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com \
+   --role=roles/artifactregistry.reader \
+   --project=<PROJECT_ID>
+```
+- Create an IAM policy binding between the Kubernetes service account and Google service account
+```angular2html
+gcloud iam service-accounts add-iam-policy-binding \
+   --role roles/iam.workloadIdentityUser \
+   --member "serviceAccount:<PROJECT_ID>.svc.id.goog[oci-webhook/webhook-server-sa]" \
+   <GSA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com \
+   --project=<PROJECT_ID>
+```
+- Annotate the Kubernetes ServiceAccount so that GKE sees the link between the service accounts
+```angular2html
+kubectl annotate serviceaccount webhook-server-sa -n oci-webhook \
+iam.gke.io/gcp-service-account=<GSA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com
+```
+
+See [Authentication to Google Cloud APIs from GKE workloads] for more details.
+
+## Deploy the webhook server and ValidatingWebhookConfiguration
+
+Replace the <PROJECT_ID> in webhook-manifest.yaml with the project name where webhook server image is hosted.
+
+Replace the <CA_BUNDLE> in the same file with the content of tls.crt `cat tls.crt | base64 -w 0`.
+
+Apply the manifest to cluster.
+
+```angular2html
+kubectl apply -f webhook-manifeset.yaml
+```
+
+## Test the image signature verification
+
+- [Install Config Sync], configure it to [sync from an unsigned OCI image].
+
+- Look for errors in webhook server log `kubectl logs deployment webhook-server -n oci-webhook`:
+
+```angular2html
+main.go:69: error during command execution: no signatures found
+```
+
+- Config Sync will also be reporting source error when running `nomos status`.
+
+```angular2html
+Error:   KNV2004: admission webhook "imageverification.webhook.com" denied the request: Image validation failed: cosign verification failed: exit status 10, output: Error: no signatures found
+main.go:69: error during command execution: no signatures found
+```
+
+- Sign the same source image using Cosign
+
+```angular2html
+cosign sign <IMAGE> --key cosign.key
+```
+
+- Once the image is correctly signed, the webhook server should successfully verify it. This will result in:
+
+  - The Config Sync source error being cleared.
+  - The configsync.gke.io/source-commit and configsync.gke.io/source-url annotations on the RootSync object being updated to reflect the new signed image.
```

Comment:
Will there be a separate user guide that explains these two new annotations and how to use them for image verification, or we need to explain them here too?

Reply:
Did add comments in the server code about the two source annotations, let me know if that's sufficient.

Reply:
There is also public documentation planned for the feature to elaborate on the new annotations.